bookkeeper-common/build.gradle

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+plugins {
+    id 'java'
+}
+
+dependencies {
+    implementation project(":bookkeeper-stats")
+    implementation project(":cpu-affinity")
+
+    compileOnly depLibs.errorprone
+    compileOnly depLibs.jsr305
+    compileOnly depLibs.lombok
+    compileOnly depLibs.spotbugsAnnotations
+    implementation depLibs.commonsConfiguration
+    implementation depLibs.guava
+    implementation depLibs.jacksonAnnotations
+    implementation depLibs.jacksonCore
+    implementation depLibs.jacksonDatabind
+    implementation depLibs.jctools
+    implementation depLibs.nettyCommon
+    implementation depLibs.slf4j
+    testCompileOnly depLibs.lombok
+    testImplementation depLibs.commonsLang3
+    testImplementation depLibs.hamcrest
+    testImplementation depLibs.junit
+    testImplementation depLibs.log4j
+    testImplementation depLibs.mockito
+    testImplementation depLibs.slf4jLog4j
+
+    annotationProcessor depLibs.lombok
+    testAnnotationProcessor depLibs.lombok
+}
+
+configurations {
+    testArtifacts.extendsFrom testRuntime
+}
+
+task testJar(type: Jar) {
+    archiveClassifier.set('test')
+    from sourceSets.test.output
+}
+
+artifacts {
+    testArtifacts testJar
+}
```

Comment:
Since Gradle 5.6, there has been java-test-fixtures plugin to achieve this, explained in https://stackoverflow.com/a/60138176 and https://docs.gradle.org/current/userguide/java_testing.html#sec:java_test_fixtures . It requires the usage of "java-library" plugin too.

Reply:
I think for this to work exposed test code have to be in bookkeeper-common/src/testFixtures directory, which is not the case. Correct me if I am wrong.
 And I can not ready to move them at least now may be later.

Reply:
yes, there's also some other work required when using the test fixtures plugin since I don't think that it's recommended to mix the actual tests with the fixtures. It's fine to go forward with the original solution.